collectors/beegfsmetaMetric.go

```diff
@@ -0,0 +1,207 @@
+package collectors
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"os/user"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	lp "github.com/ClusterCockpit/cc-metric-collector/internal/ccMetric"
+)
+
+// Struct for the collector-specific JSON config
+type BeegfsMetaCollectorConfig struct {
+	Beegfs            string   `json:"beegfs_path"`
+	ExcludeMetrics    []string `json:"exclude_metrics,omitempty"`
+	ExcludeFilesystem []string `json:"exclude_filesystem"`
+}
+
+type BeegfsMetaCollector struct {
+	metricCollector
+	tags    map[string]string
+	matches map[string]string
+	config  BeegfsMetaCollectorConfig
+	skipFS  map[string]struct{}
```

Comment:
Why do you use `struct{}` here? I (as Golang novice like you) use `bool` for such simple lookup tables.

Reply:
This lines where clones from other Collectors. Other Golang experts say, that this way some memory can be saved.

Reply:
Only the GpfsCollector uses `struct{}`. I'm not sure about the memory argument, a `bool` might be a single bit while any pointer needs at least some bytes.